core/src/impl/KokkosExp_Host_IterateNestLoopWoTile.hpp

```diff
@@ -0,0 +1,598 @@
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright Contributors to the Kokkos project
+
+#ifndef KOKKOS_HOST_EXP_ITERATE_NESTLOOPWOTILE_HPP
+#define KOKKOS_HOST_EXP_ITERATE_NESTLOOPWOTILE_HPP
+
+#include <Kokkos_Macros.hpp>
```

Comment:
```suggestion
#include <type_traits>

#include <Kokkos_Macros.hpp>
#include <Kokkos_Layout.hpp>
```
It compiles fine as is, but this relies on the right headers (`<type_traits>` for `enable_if` and `Kokkos_Layout.hpp` for `Iterate::Left`) to be included before this file is included, which is not the cleanest thing to do imo

Reply:
I have taken care of this comment. Squashing and force pushing my changes at this stage.